cmd/end2endtest/encrypted.go

```diff
@@ -0,0 +1,381 @@
+		Description: map[string]string{"default": "Test election description"},
+		EndDate:     time.Now().Add(time.Minute * 20),
+
+		VoteType: vapi.VoteType{
+			UniqueChoices:     false,
+			MaxVoteOverwrites: 1,
+		},
+
+		ElectionType: vapi.ElectionType{
+			Autostart:         true,
+			Interruptible:     true,
+			Anonymous:         false,
+			SecretUntilTheEnd: true,
+			DynamicCensus:     false,
+		},
+
+		Census: vapi.CensusTypeDescription{
+			RootHash: root,
+			URL:      censusURI,
+			Type:     "weighted",
+			Size:     uint64(len(voterAccounts)),
+		},
+
+		Questions: []vapi.Question{
+			{
+				Title:       map[string]string{"default": "Test question 1"},
+				Description: map[string]string{"default": "Test question 1 description"},
+				Choices: []vapi.ChoiceMetadata{
+					{
+						Title: map[string]string{"default": "Yes"},
+						Value: 0,
+					},
+					{
+						Title: map[string]string{"default": "No"},
+						Value: 1,
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Infof("created new election with id %s - now wait until it starts", electionID.String())
+
+	// Wait for the election to start
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*40)
+	defer cancel()
+	election, err := api.WaitUntilElectionStarts(ctx, electionID)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Debugf("election details: %+v", *election)
+
+	// Send the votes (parallelized)
+	startTime := time.Now()
+	wg = sync.WaitGroup{}
+	voteAccounts := func(accounts []*ethereum.SignKeys, wg *sync.WaitGroup) {
+		defer wg.Done()
+		log.Infof("sending %d votes", len(accounts))
+		// We use maps instead of slices to have the capacity of resending votes
+		// without repeating them.
+		accountsMap := make(map[int]*ethereum.SignKeys, len(accounts))
+		for i, acc := range accounts {
+			accountsMap[i] = acc
+		}
+		// Send the votes
+		votesSent := 0
+		for {
+			contextDeadlines := 0
+			for i, voterAccount := range accountsMap {
+				c := api.Clone(fmt.Sprintf("%x", voterAccount.PrivateKey()))
+				_, err := c.Vote(&apiclient.VoteData{
+					ElectionID:  electionID,
+					ProofMkTree: proofs[voterAccount.Address().Hex()],
+					Choices:     []int{i % 2},
+				})
+				// if the context deadline is reached, we don't need to print it (let's just retry)
+				if err != nil && errors.Is(err, context.DeadlineExceeded) || os.IsTimeout(err) {
+					contextDeadlines++
+					continue
+				} else if err != nil && !strings.Contains(err.Error(), "already exists") {
+					// if the error is not "vote already exists", we need to print it
+					log.Warn(err)
+					continue
+				}
+				// if the vote was sent successfully or already exists, we remove it from the accounts map
+				votesSent++
+				delete(accountsMap, i)
+
+			}
+			if len(accountsMap) == 0 {
+				break
+			}
+			log.Infof("sent %d/%d votes... got %d HTTP errors", votesSent, len(accounts), contextDeadlines)
```

Comment:
I think this log maybe should before the break in order to show the vote information of each chunk of accounts passed to the function

Reply:
true, the contextDeadlines is not reported in the line below
("successfully sent %d votes")
but i guess it's not so important in case of success. so i'll leave it as is